### PR TITLE
refactoring code according to class diagram

### PR DESCRIPTION
**Pull request by sanjaythiru**
refactoring code according to class diagram

### CODE REVIEW

**Comment by sanjaythiru** [comment]
I have not added AbstractService and AbstractClient as those could be added when we implement ChessComService and ChessComClient

**Comment by sanjaythiru** [comment]
I will add the unit tests soon.
Please share your comments.
I can work on addressing it.
Also, please feel free to merge if you are good with the changes.

**Comment by sanjaythiru** [comment]
Next task would be to assign clubscore according to points and tiebreak.
I have added a TODO for that.

**Comment by sanjaythiru** [comment]
Currently, we are printing in console.
Its okay. 
I also want to add more print statements / logging so that we can see what is getting executed.

**Comment by sanjaythiru** [comment]
I have added a commit with logging.
Please let me know your thoughts.

File: src/main/java/com/sreeram/bccleaderboard/client/LichessClient.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        if (lastFiveSwissTournaments instanceof Entries<Swiss> entries) {
+            List<Swiss> tmtEntries = entries.stream().toList();
+            LOGGER.info("{} Request success : Club {}, Count {}, response {}", LICHESS, club, count, tmtEntries);
+            if (tmtEntries.isEmpty()) {
+                LOGGER.warn("Empty tournament list");
+                return new ArrayList<>();
+            }
+            return tmtEntries.stream().map(this::convert).toList();
+        } else {
+            LOGGER.error("{} Request failed : Club {}, Count {}", LICHESS, club, count);
```
Comment (sanjaythiru) [inline src/main/java/com/sreeram/bccleaderboard/client/LichessClient.java:38]: Log failure message status and info

File: src/main/java/com/sreeram/bccleaderboard/client/LichessClient.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            LOGGER.error("{} Request failed : top 10 players - Tournament {}, failure message {}", LICHESS, tmt.getId(), topTen);
+            return new ArrayList<>();
+        }
+    }
+
+    private Tournament convert(Swiss lichessTmt) {
+        LOGGER.debug("Converting Swiss -> Tournament, tmtId {}", lichessTmt.id());
+        Tournament tournament = new Tournament(lichessTmt.id());
+        tournament.setStartsAt(OffsetDateTime.parse(lichessTmt.startsAt()));
+        tournament.setStatus(lichessTmt.status());
```
Comment (sanjaythiru) [inline src/main/java/com/sreeram/bccleaderboard/client/LichessClient.java:62]: Filter swiss tournaments based on finished status in convert method itself

Comment (sanjaythiru) [reply src/main/java/com/sreeram/bccleaderboard/client/LichessClient.java:62]: Remove status from Tournament

Comment (sanjaythiru) [reply src/main/java/com/sreeram/bccleaderboard/client/LichessClient.java:62]: Decision - convert method only for Dto to Bo.
No logic should be implemented in convert method

File: src/main/java/com/sreeram/bccleaderboard/models/Player.java [hunk 3]
```diff
@@ -1,23 +1,58 @@
...
 public class Player {
-  private String username;
-  private String monthScore;
-  
-  public String getMonthScore() {
-    return monthScore;
-  }
+  private final String username;
+
+  // I feel combining the below tournament related details together would be good.
```
Comment (sanjaythiru) [inline src/main/java/com/sreeram/bccleaderboard/models/Player.java:12]: Create an private inner class with tmtPoint, tiebreak and clubScore

**Review by sreeramk28 — APPROVED** [review]
Looks good da. We will work on the things we have commented post merge. Thank you

**Pull request merged**